### PR TITLE
TG-102: Employee list, create and edit shared components implemented

### PR DESCRIPTION
**Pull request by LucasMattia**
Done alongside @SuarezAlexisEzequiel

### CODE REVIEW

File: src/Components/Employees/FormEmployee/FormEmployee.jsx [hunk 1]
```diff
@@ -17,27 +23,18 @@ const FormEmployee = (props) => {
...
-      ) {
-        alert('Employee created');
-        props.history.push('/employees');
-      } else {
-        alert('Please fill every field');
-      }
+      setStatus(response.status);
+      setModalText(createMsg(response.status));
+      setIsOpen(true);
+      console.log(response);
```
Comment (cocolobos) [inline src/Components/Employees/FormEmployee/FormEmployee.jsx:37]: remove console log

File: src/Components/Employees/FormEmployee/FormEmployee.jsx [hunk 2]
```diff
@@ -17,27 +23,18 @@ const FormEmployee = (props) => {
...
-        employeeInput.password !== '' &&
-        employeeInput.active !== ''
-      ) {
-        alert('Employee created');
-        props.history.push('/employees');
-      } else {
-        alert('Please fill every field');
-      }
+      setStatus(response.status);
+      setModalText(createMsg(response.status));
```
Comment (cocolobos) [inline src/Components/Employees/FormEmployee/FormEmployee.jsx:35]: can we remove the status on the modal ?

File: src/Components/Employees/FormEmployee/FormEmployee.jsx [hunk 3]
```diff
@@ -56,53 +53,83 @@ const FormEmployee = (props) => {
...
-            <input type="text" name="active" value={employeeInput.active} onChange={onChange} />
-          </div>
-          <div>
-            <input type="submit" value="submit" />
-          </div>
-        </form>
-      </div>
-      <button onClick={() => props.history.push('/employees')}>Return</button>
+      <Modal showModal={isOpen} closeModal={() => setIsOpen(false)}>
+        <h2>Status: {status}</h2>
```
Comment (cocolobos) [inline src/Components/Employees/FormEmployee/FormEmployee.jsx:123]: remove status from confirmation message

File: src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx [hunk 4]
```diff
@@ -121,11 +147,24 @@ const FormEmployeeEdit = (props) => {
...
+            <input className={styles.addEmployeeBtn} type="submit" value="submit" />
           </div>
         </form>
+        <Button type={styles.addEmployeeBtn} handleClick={() => props.history.push('/employees')}>
+          return
+        </Button>
       </div>
-      <button onClick={() => props.history.push('/employees')}>Return</button>
+      <Modal showModal={isOpen} closeModal={() => setIsOpen(false)}>
+        <h2>Status: {status}</h2>
```
Comment (cocolobos) [inline src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx:158]: same here, delete the status

File: src/Components/Employees/index.js [hunk 5]
```diff
@@ -20,22 +28,45 @@ const Employees = (props) => {
...
-      </div>
+      <Logo />
+      <h2 className={styles.title}>Employees</h2>
+      <Table
+        data={list}
+        headers={['firstName', 'lastName', 'phone', 'email', 'password', 'active']}
+        handleEdit={handleEdit}
+        deleteItem={deleteItem}
+      ></Table>
+      <Button
```
Comment (Lucas-DeLorenzi) [inline src/Components/Employees/index.js:53]: this button should be displayed above the table

File: src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx [hunk 6]
```diff
@@ -121,11 +147,24 @@ const FormEmployeeEdit = (props) => {
...
+        <div>
+          <p>{checkEmployee(modalText)}</p>
+        </div>
+        <div>
+          <Button type={styles.addEmployeeBtn} handleClick={() => props.history.push('/employees')}>
```
Comment (Lucas-DeLorenzi) [inline src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx:163]: this button should only return us to /employees on success case

File: src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx [hunk 7]
```diff
@@ -59,8 +62,31 @@ const FormEmployeeEdit = (props) => {
...
+      result = '400';
+    }
+
+    return result;
+  };
+
+  const checkEmployee = () => {
+    let result;
+    if (!modalText) {
+      result = 'There was an error';
```
Comment (Lucas-DeLorenzi) [inline src/Components/Employees/FormEmployee/FormEmployeeEdit.jsx:79]: we could change "there was an error" to "the fields were not filled in correctly, please check the data" or something like that

File: src/Components/Employees/employees.module.css [hunk 8]
```diff
@@ -2,14 +2,37 @@
   flex: 1;
   display: flex;
   flex-direction: column;
-  justify-content: center;
+  justify-content: space-between;
   align-items: center;
+  padding-top: 10px;
+}
+
+.title{
```
Comment (Lucas-DeLorenzi) [inline src/Components/Employees/employees.module.css:10]: the color chould be #373867 following figma design

**Review by JulianDeme — APPROVED** [review]

**Pull request merged**